client/blocks/credit-card-form/index.jsx

```diff
@@ -305,40 +302,29 @@ const CreditCardForm = createReactClass( {
 					</div>
 					{ this.renderUsedForExistingPurchases() }
 				</Card>
-
 				<CompactCard className="credit-card-form__footer">
-					<em>{ this.props.translate( 'All fields required' ) }</em>
-
+					<em>{ translate( 'All fields required' ) }</em>
 					<FormButton disabled={ this.state.formSubmitting } type="submit">
-						{ this.state.formSubmitting
-							? this.props.translate( 'Saving Card…', {
-									context: 'Button label',
-									comment: 'Credit card',
-								} )
-							: this.props.translate( 'Save Card', {
-									context: 'Button label',
-									comment: 'Credit card',
-								} ) }
+						{ this.getSubmitButtonText() }
 					</FormButton>
 				</CompactCard>
 			</form>
 		);
-	},
+	}
 
 	renderUsedForExistingPurchases() {
-		if ( this.props.showUsedForExistingPurchasesInfo ) {
+		const { translate, showUsedForExistingPurchasesInfo } = this.props;
+		if ( showUsedForExistingPurchasesInfo ) {
```

Comment:
it would be nice to refactor this into an early-abort pattern instead of nesting…

```js
if ( ! showUsedForExistingPurchasesInfo ) {
	return;
}

return (
	<div …>
		<Gridicon … />
		…
	</div>
)
```